nntrainer/graph/network_graph.cpp

```diff
@@ -576,8 +576,22 @@ void NetworkGraph::inPlaceOptimize(const std::string &layer_type,
         continue;
 
       /** Share tensor with next layer */
-      l->net_input[0] = l->net_hidden[0];
-      prev_layer->net_hidden[loc] = l->net_hidden[0];
+      l->net_input[0] = l->net_hidden[0]; // I2 = O2
+      if (l->getType() == BatchNormalizationLayer::type) {
```

Comment:
caution for this if,
we will going to need `is_batchnorm_type`, `is_activation_type` kind of property if this cannot be unified.

Reply:
We could have strategy instead I guess?

Reply:
The strategy is to support `supportInPlace()` in the layer.

Reply:
#867